fdbclient/FileBackupAgent.actor.cpp

```diff
@@ -1091,27 +1098,30 @@ ACTOR static Future<Void> decodeKVPairs(StringRefReader* reader,
 
 		// make sure that all keys in a block belong to exactly one tenant,
 		// unless its the last key in which case it can be a truncated (different) tenant prefix
-		ASSERT(!encryptedBlock || blockDomainId.present());
-		if (CLIENT_KNOBS->SIMULATION_ENABLE_SNAPSHOT_ENCRYPTION_CHECKS && encryptedBlock && g_network &&
-		    g_network->isSimulated() && !isReservedEncryptDomain(blockDomainId.get())) {
+		if (encryptedBlock && g_network && g_network->isSimulated()) {
+			ASSERT(blockDomainId.present());
 			state KeyRef curKey = KeyRef(k, kLen);
 			if (!prevDomainId.present()) {
-				EncryptCipherDomainId domainId =
-				    wait(EncryptedRangeFileWriter::getEncryptionDomainDetails(prevKey, encryptMode, tenantCache));
+				EncryptCipherDomainId domainId = wait(
+				    EncryptedRangeFileWriter::getEncryptionDomainDetails(prevKey, encryptMode, tenantCache, false));
 				prevDomainId = domainId;
 			}
-			EncryptCipherDomainId curDomainId =
-			    wait(EncryptedRangeFileWriter::getEncryptionDomainDetails(curKey, encryptMode, tenantCache));
+			state EncryptCipherDomainId curDomainId =
+			    wait(EncryptedRangeFileWriter::getEncryptionDomainDetails(curKey, encryptMode, tenantCache, false));
 			if (!curKey.empty() && !prevKey.empty() && prevDomainId.get() != curDomainId) {
 				ASSERT(!done);
-				if (curDomainId != SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID && curDomainId != FDB_DEFAULT_ENCRYPT_DOMAIN_ID) {
-					ASSERT(curKey.size() == TenantAPI::PREFIX_SIZE);
+				// Make sure that all tenant specific keys in a block have the correct prefix size
+				if (curDomainId != SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID && curDomainId != FDB_DEFAULT_ENCRYPT_DOMAIN_ID &&
```

Comment:
What happens to the KV where `Key.size() != TENANT_PREFIX`?

Reply:
For System and Default domains we dont truncate the last key in the block. Also if it was a first/last key in a file (see above comment) then we will also not truncate the key

Reply:
So how would encryption work in such cases? Do we not encrypt this file or block?

Reply:
Yes its encrypted with the default key